### DISCUSSION TITLE
Additional Snippets

### DISCUSSION THREAD

**Discussion by BayBreezy** [nuxtrdev/nuxtr-vscode#27] [Ideas]
Hello 👋,

Thank you for the extension. I have it installed in my Nuxt 3 profile for VSCode.

I have a suggestion for the snippets shipped with this extension. It is around the common TS definitions for the different macros available in Vue 3. When I start a new project, I usually create a snippet file for that project and add the snippets below. Can they be added to the core extension?

```json
  "create.props": {
    "scope": "javascript,typescript",
    "description": "Create props block in vue 3",
    "prefix": "props",
    "body": ["const props = defineProps<{", "  $0", "}>();"]
  },
  "create.defaultprops": {
    "scope": "javascript,typescript",
    "description": "Create props block with defaults in vue 3",
    "prefix": "propsdf",
    "body": ["const props = withDefaults(defineProps<{", "  $0", "}>(),{});"]
  },
  "create.defineoptions": {
    "scope": "javascript,typescript",
    "description": "Create define options in setup",
    "prefix": "defineoptions",
    "body": ["defineOptions({ $0 });"]
  },
  "Create new page": {
    "scope": "html, vue",
    "prefix": "page",
    "body": [
      "<template>",
      "\t<div>",
      "",
      "\t</div>",
      "</template>",
      "",
      "<script setup lang=\"ts\">",
      "",
      "</script>",
      ""
    ]
  }
```

Let me know what you think :)

**Comment by adhamfarrag** [comment]
Hello @BayBreezy :)

Thanks for your kind words. Nice that you have a dedicated VSCode profile for that :) 

- To maintain Nuxtr scope, we decided not to support Vue snippets. At least not with Nuxtr VSCode.
- Unless you need to share it with a team. I believe no need to create these snippets on a project level, if you're interested you can check [this reference.](https://code.visualstudio.com/docs/editor/userdefinedsnippets#_create-your-own-snippets)
- Nuxtr already offers all the controllers you need over Vue File templates. No need to include the last snippet in your project snippets. More about this [here](https://github.com/nuxtrdev/nuxtr-vscode#project-directories-and-files-creation).

Next release should have major updates and features for Snippets. Can't wait to share it with you.

**Comment by BayBreezy** [comment]
Okie cool. Looking forward to it :)

**Comment by adhamfarrag** [comment]
@BayBreezy Hey, I tried DMing you on X but your DM settings doesn't allow me. Came across your [recent tutorial](https://www.youtube.com/watch?v=qzTXEvXpuYo) and made me curious to chat little bit with you about Nuxtr for new learner from your perspective as a tutor. Please if you're interested, my X account is on my Github profile and DMs should be open. Thanks :)
